jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -0,0 +1,415 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.victools.jsonschema.plugin.maven;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.victools.jsonschema.generator.Module;
+import com.github.victools.jsonschema.generator.Option;
+import com.github.victools.jsonschema.generator.OptionPreset;
+import com.github.victools.jsonschema.generator.SchemaGenerator;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfig;
+import com.github.victools.jsonschema.generator.SchemaGeneratorConfigBuilder;
+import com.github.victools.jsonschema.generator.SchemaVersion;
+import com.github.victools.jsonschema.module.jackson.JacksonModule;
+import com.github.victools.jsonschema.module.jackson.JacksonOption;
+import com.github.victools.jsonschema.module.javax.validation.JavaxValidationModule;
+import com.github.victools.jsonschema.module.javax.validation.JavaxValidationOption;
+import com.github.victools.jsonschema.module.swagger15.SwaggerModule;
+import com.github.victools.jsonschema.module.swagger15.SwaggerOption;
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * Maven plugin for the victools/jsonschema-generator.
+ */
+@Mojo(name = "schema-generator",
+        defaultPhase = LifecyclePhase.PROCESS_TEST_SOURCES,
+        requiresDependencyResolution = ResolutionScope.RUNTIME)
+public class SchemaGeneratorMojo extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
+
+    /**
+     * Full name of the class for which the JSON schema will be generated.
+     */
+    @Parameter(property = "className", required = true)
+    private String className;
```

Comment:
What changes would be necessary to allow for multiple schemas to be generated?

Reply:
- We could make the className a collection type, so you can specify multiple classes.
In that case we need to think about a naming convention for the generated schema files when the default file name is not good enough.
What is actually a good naming convention for JSON schema files?
- There was also a request to use all classes in a package/folder. We would need to investigate how we can find all classes in a package/folder.

Reply:
- Having both options side-by-side (a single `className` and a `classNames` collection) sounds good to me.
- I've no idea what a common naming convention for schema files is. Your proposed `*.schema.json` sounds good enough to me (except for maybe an underscore before the `schema` part? `*_schema.json`).
- As to the file name configuration, maybe we could have a nested structure then:
```xml
<configuration>
    <targetTypes>
        <targetType>
            <className>a.b.c.YourClass</className>
        </targetType>
        <targetType>
            <className>a.b.c.MyClass</className>
            <schemaFileName>MyClass_schema.json</schemaFileName>
        </targetType>
    </targetTypes>
    <targetPackages>
        <targetPackage>x.y.z</targetPackage>
    </targetPackages>
    <schemaFilePath>src/main/resources/schema</schemaFilePath>
</configuration>
```
- I've been using `org.reflections:reflections` in the past for things like looking-up all classes from a certain package. It's easy enough to use and introducing an extra dependency shouldn't be an issue for this maven plugin. I'd also find it acceptable if the file names cannot be configured for this kind of per-package schema generation.